src/utils/user.js

```diff
@@ -1,17 +1,33 @@
+import { Base64 } from "js-base64";
 import asyncStore from "./asyncstorage";
+import authServices from "services/auth";
 
-async function checkIfLoggedIn() {
+async function checkForUserLogin() {
+  let user;
   try {
-    const user = JSON.parse(await asyncStore.retrieve("user"));
-    if (user && user.expireAt && user.expireAt > Date.now()) {
-      return user;
+    user = JSON.parse(await asyncStore.retrieve("user"));
+    if (!user) return null;
+    if (user instanceof Error) throw user;
+    if (user && !user.token) throw new Error("No user login token.");
+
+    // Check if header has valid info
+    const tokenHeader = JSON.parse(Base64.decode(user.token.split(".")[0]));
+    if (!tokenHeader.kid) throw new Error("Missing token key id.");
+    if (tokenHeader.kid !== user.username) {
+      throw new Error("Login token key id did not match username.");
```

Comment:
Is there a library that allows us to validate the token? I'm surprised that we have to do this much manual checking ourselves!

Reply:
Token validation happens on the server side, but this is here to make sure that the token saved on the client device matches up with the user the client thinks they currently are. It's not technically necessary but it's useful if the token saving goes awry-- don't want to allow the user to create or update an event as another user.

Reply:
OK got it!